tsconfig.base.json

```diff
@@ -12,7 +12,7 @@
     "moduleResolution": "node",
     "resolveJsonModule": true,
     "noImplicitAny": false,
-    "outDir": "../../js"
+    "outDir": "./js"
   },
-  "exclude": ["../../utils"]
+  "exclude": ["utils", "build", "**/test_common", "**/*.js"]
```

Comment:
As this is being extended by `/client` and `/server` - are the relative paths not needed? (Including `outdir`)

And why are we ignoring `.js` files? I thought we were using `allowJS` to ensure we transpile the compiled JS + any source JS to the right es module support?

Reply:
At one stage I ran into an issue with the path config, show did a showConfig tsc run to debug. There I saw all utility, build, test code and linter (symlinks) + jest config being referenced in both the client and server outputs. I don't know how, but it seems when extended, the exclude is witth respect to where the config that defines it is. Hence the traversal didnt work.

This is an additional change - one I can back out. But with it, The config looked as expected for the client and server

Reply:
I trust you :)